kotlinx-coroutines-core/common/src/channels/AbstractChannel.kt

```diff
@@ -112,9 +112,10 @@ internal abstract class AbstractSendChannel<E> : SendChannel<E> {
         queue: LockFreeLinkedListHead,
         element: E
     ) : AddLastDesc<SendBuffered<E>>(queue, SendBuffered(element)) {
-        override fun failure(affected: LockFreeLinkedListNode, next: Any): Any? {
-            if (affected is ReceiveOrClosed<*>) return OFFER_FAILED
-            return null
+        override fun failure(affected: LockFreeLinkedListNode, next: Any): Any? = when (affected) {
```

Comment:
`next` parameter is no longer used by production code.
It is only used in `describeRemove` for tests

Reply:
Removed `next`, removed `LockFreeLinkedList.describeRemove` (as not really needed in production) and the corresponding tests.